hbase_sql/hbase1.1.1/src/main/java/com/splicemachine/hbase/BackupEndpointObserver.java

```diff
@@ -138,6 +138,18 @@ public void prepareBackup(
         if (isSplitting.get() || isFlushing.get() || isCompacting.get()) {
             SpliceLogUtils.info(LOG, "table %s region %s is not ready for backup: isSplitting=%s, isCompacting=%s, isFlushing=%s",
                     tableName , regionName, isSplitting.get(), isCompacting.get(), isFlushing.get());
+
+            // If the region is flush, wait until it is done. Check the flag. If it is still true and the region
+            // gets stuck in split state, flush it to reset the flag
+            if (isFlushing.get()) {
+                region.waitForFlushesAndCompactions();
+                if (isFlushing.get()) {
+                    SpliceLogUtils.warn(LOG, "The region %s may be stuck in split state, flush it to reset flag",
+                            regionName);
+                    HBasePlatformUtils.flush(region);
+                }
```

Comment:
If we're waiting for the flushes and compactions to complete, why do we still fall through and call responseBuilder.setReadyForBackup(false) ?  Could we just re-apply the splitting/flushing/compacting test here?, e.g.

if (!(isSplitting.get() || isFlushing.get() || isCompacting.get()))
    return super.preCompact(e, store, scanner, scanType);

Reply:
Yes, we could, but I prefer to make simple code changes here. This is to handle a rare case. Returning false and letting backup(for this region) retry does not have a significant impact on performance.

Reply:
OK, I didn't realize if we return false, the backup will get retried.  It seems fine.  Thanks.